src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -1659,49 +1665,58 @@ public ILocation getRequesterLocation()
+        if (getOpenRequestsByCitizen().get(citizenThatRequested).isEmpty())
         {
-            citizensByRequests.remove(citizenThatRequested);
+            getOpenRequestsByCitizen().remove(citizenThatRequested);
         }
 
         final IRequest<?> requestThatCompleted = getColony().getRequestManager().getRequestForToken(token);
-        openRequests.get(TypeToken.of(requestThatCompleted.getRequest().getClass())).remove(token);
+        getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).remove(token);
 
-        if (openRequests.get(TypeToken.of(requestThatCompleted.getRequest().getClass())).isEmpty())
+        if (getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).isEmpty())
         {
-            openRequests.remove(TypeToken.of(requestThatCompleted.getRequest().getClass()));
+            getOpenRequestsByRequestableType().remove(TypeToken.of(requestThatCompleted.getRequest().getClass()));
         }
 
-        if (!citizensByCompletedRequests.containsKey(citizenThatRequested))
+        if (!getCompletedRequestsByCitizen().containsKey(citizenThatRequested))
         {
-            citizensByCompletedRequests.put(citizenThatRequested, new ArrayList<>());
+            getCompletedRequestsByCitizen().put(citizenThatRequested, new ArrayList<>());
         }
-        citizensByCompletedRequests.get(citizenThatRequested).add(token);
+        getCompletedRequestsByCitizen().get(citizenThatRequested).add(token);
 
         markDirty();
     }
 
-    public void onRequestCancelled(@NotNull final IToken<?> token)
+    @Override
+    @NotNull
+    public void onRequestCancelled(@NotNull final IRequestManager manager, @NotNull final IToken token)
     {
-        final Integer citizenThatRequested = requestsByCitizen.remove(token);
-        citizensByRequests.get(citizenThatRequested).remove(token);
+        final int citizenThatRequested = getCitizensByRequest().remove(token);
+        getOpenRequestsByCitizen().get(citizenThatRequested).remove(token);
 
-        if (citizensByRequests.get(citizenThatRequested).isEmpty())
+        if (getOpenRequestsByCitizen().get(citizenThatRequested).isEmpty())
         {
-            citizensByRequests.remove(citizenThatRequested);
+            getOpenRequestsByCitizen().remove(citizenThatRequested);
         }
 
         final IRequest<?> requestThatCompleted = getColony().getRequestManager().getRequestForToken(token);
-        openRequests.get(TypeToken.of(requestThatCompleted.getRequest().getClass())).remove(token);
-
-        if (openRequests.get(TypeToken.of(requestThatCompleted.getRequest().getClass())).isEmpty())
+        if (requestThatCompleted != null)
         {
-            openRequests.remove(TypeToken.of(requestThatCompleted.getRequest().getClass()));
+            if (getOpenRequestsByRequestableType().containsKey(TypeToken.of(requestThatCompleted.getRequest().getClass())))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)